operator/v1/types_ingress.go

```diff
@@ -925,6 +936,32 @@ type IngressControllerHTTPHeaders struct {
 	HeaderNameCaseAdjustments []IngressControllerHTTPHeaderNameCaseAdjustment `json:"headerNameCaseAdjustments,omitempty"`
 }
 
+// IngressControllerHTTPHeaderBuffer specifies the size of the
+// per-connection HTTP header buffers.
+type IngressControllerHTTPHeaderBuffer struct {
+	// headerBufferBytes describes how much memory should be reserved
+	// (in bytes) for IngressController connection sessions.
+	// Note that this value must be at least 16384 if HTTP/2 is
+	// enabled for the IngressController (https://tools.ietf.org/html/rfc7540).
+	//
+	// +kubebuilder:validation:Required
+	// +kubebuilder:validation:Minimum=16384
+	// +required
+	HeaderBufferBytes int32 `json:"headerBufferBytes"`
+
+	// headerBufferMaxRewriteBytes describes how much memory should be reserved
+	// (in bytes) from headerBufferBytes for HTTP header rewriting
+	// and appending for IngressController connection sessions.
+	// Note that incoming HTTP requests will be limited to
+	// (headerBufferBytes - headerBufferMaxRewriteBytes) bytes, meaning
+	// headerBufferBytes must be greater than headerBufferMaxRewriteBytes.
+	//
+	// +kubebuilder:validation:Required
+	// +kubebuilder:validation:Minimum=4096
+	// +required
```

Comment:
so the intent is, "if the struct is set, then both of these values must be set"?  I expected both of these values to be optional to allow independent specification.

Reply:
Yes, this is the intent. In general, `HeaderBufferBytes` scale with `HeaderBufferMaxRewriteBytes`.